tests/page-object-model/loginpage.ts

```diff
@@ -0,0 +1,49 @@
+import { Page, Locator, expect } from "@playwright/test";
+import { BasePage } from "./basepage";
+
+export class LoginPage extends BasePage {
+    readonly expect: LoginPageAssertions
+    readonly _usernameFieldLocator:Locator;    
+    readonly _passwordFieldLocator: Locator;
+    readonly _submitButtonFieldLocator: Locator;
+
+    constructor(page: Page) {
+        super(page);
+        this.expect = new LoginPageAssertions(this);
+
+        this._usernameFieldLocator = this.page.locator('#username');
+        this._passwordFieldLocator = this.page.locator('#password');
+        this._submitButtonFieldLocator = this.page.locator('"Login"');
+    }
+
+    override async navigateTo(): Promise<void> {
+        await super.navigateTo('/login');
+    }
+
+    async submitForm(): Promise<void> {
+        await this._submitButtonFieldLocator.click();        
+    }
+
+    async fillFormWithValidDetails(): Promise<void> {
+        await this._usernameFieldLocator.fill("name");
+        await this._passwordFieldLocator.fill("pass");
```

Comment:
Nitpick: No biggie but could pop the creds in a .env file

Reply:
and could also include include what you need to set in the local storage in a .env i.e;
```
UI_AUTH_KEY=authenticated
UI_AUTH_VALUE=true
```

Reply:
I absolutely agree :-)
Config is on my list of things to do - atm its all hardcoded!